src/bindings/g_object/object.cr

```diff
@@ -508,6 +508,12 @@ module GObject
         end
 
         def emit({{ signature.args.splat }})
+          {% for arg in signature.args %}
+            {% resolved_type = arg.type.resolve %}
+            {% if resolved_type < Reference && resolved_type != ::String && !(resolved_type < ::GObject::Object) %}
```

Comment:
Does this work with `GObject.ParamSpec`?
Maybe another test case.

Reply:
good catch, I think we need to check if any ancestors have the generator annotation, so we know that it’s a type gi-crystal can rebuild and is safe to use `to_unsafe`.